Source/monster.cpp

```diff
@@ -543,12 +541,14 @@ void PlaceUniqueMonsters()
 		if (UniqueMonstersData[u].mlevel != currlevel)
 			continue;
 
-		int mt;
-		for (mt = 0; mt < LevelMonsterTypeCount; mt++) {
-			if (LevelMonsterTypes[mt].mtype == UniqueMonstersData[u].mtype)
-				break;
-		}
-		if (mt >= LevelMonsterTypeCount)
+		int mt = [&]() {
+			for (int i = 0; i < LevelMonsterTypeCount; i++) {
+				if (LevelMonsterTypes[mt].mtype == UniqueMonstersData[u].mtype)
+					return i;
+			}
+			return LevelMonsterTypeCount;
+		}();

```

Comment:
This looks identical to the block in `AddMonsterType`, maybe make them in to named functions and take type as an input. `int GetMonsterTypeIndex(uint8_t type)`